x-pack/plugins/ml/public/application/services/anomaly_detector_service.ts

```diff
@@ -7,53 +7,32 @@
 
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
-import { Job, JobId } from '../../../common/types/anomaly_detection_jobs';
-import { basePath } from './ml_api_service';
+import type { Job, JobId } from '../../../common/types/anomaly_detection_jobs';
 import { HttpService } from './http_service';
+import { type MlApiServices, mlApiServicesProvider } from './ml_api_service';
 
 export class AnomalyDetectorService {
-  private readonly apiBasePath = basePath() + '/anomaly_detectors';
+  private mlApiServices: MlApiServices;
 
-  constructor(private httpService: HttpService) {}
+  constructor(httpService: HttpService) {
```

Comment:
I've updated this class to remove the API calls directly using `http`.
It now uses our standard api service where I've added a observable based version of `getJobs`.

I've also removed the `extractInfluencers` method as has no dependencies on class member data and so can live with our other job utils functions.